[FE] 알림 기능 구현

ChanYangYu opened the pull request:
## 관련 이슈

- closes #208

## 셀프 체크리스트

> 최소 1명 이상의 assign을 받아야만 Merge 가능합니다.

- [ ] Warning Message 발생 유/무
- [ ] Coding Convention 준수 유/무
- [ ] test 통과 여부

## 작업 요약
> 작업 내역 요약

- 알람 모달 구현
## Preview

> 선택 사항



https://user-images.githubusercontent.com/55623688/143688064-00363544-ed3f-4ab0-a13e-334d14e8dab6.mp4




## PR 특이 사항

> PR을 볼 때 주의깊게 봐야하거나 말하고 싶은 점
> 백로그에 없지만 필요하여 구현한 기능 

- 이제 API랑 연결하면 될것같습니다!

ChanYangYu committed:
[FE] feat: 알림 메세지 박스 기본 레이아웃 마크업 


- issue (#208)

ChanYangYu committed:
[FE] feat: 알림 리스트 출력 및 메세지 구현


- 타입에 따라 메세지가 분기되어 출력되도록 구현
- issue (#208)

ChanYangYu committed:
[FE] feat: 알람이 생성된 시간 변환 유틸 함수 구현


- issue (#208)

ChanYangYu committed:
[FE] feat: 알림 전역 상태관리 슬라이스 추가


- 서버데이터 타입 정의 및 전역상태관리 슬라이스 추가
- issue (#208)

ChanYangYu committed:
[FE] fix: 알림 모달이 안사라지는 오류 수정


- 알림모달이 켜져있는 상태에서 프로필 버튼 클릭시 모달이 꺼지지 않고 중복해서 켜지는 문제 해결
- issue (#208)

ChanYangYu committed:
[FE] feat: 알림 전역 데이터 적용


- Notification 컴포넌트에서 더미데이터 삽입
- 삽입된 더미 데이터 데이터 출력되도록 구현
- issue (#208)

ChanYangYu committed:
[FE] feat: 삭제 기능 구현 


- 삭제 버튼 클릭시 전역상태 변경
- issue (#208)

ChanYangYu committed:
[FE] feat: 해당 알림과 연결된 링크 구현


- 그룹, 멘토 수락 알림 클릭시 그룹목록 페이지로 이동
- 멘토신청 알림 클릭시 프로필페이지 이동
- 그룹 참가 신청알림 클릭시 그룹장페이지로 이동

ChanYangYu committed:
[FE] feat: 열람한 알림 색상 변경 


- 이미 열람한 알림 색상 흐리게 표시
- 삭제버튼 호버링 효과 추가
- issue (#208)

ChanYangYu committed:
[FE] feat: 새로운 알림 표시 기능 구현 


- 아직 읽지않은 알림이 있을 경우 아이콘 옆에 주황색 알림 표시
- 모두 클릭해서 확인할 경우 알림 제거
- 기타 디자인 개선
- issue (#208)

Donggggg commented on client/src/components/BubbleModal/Notification/EmptyItem.tsx:
```diff
@@ -0,0 +1,32 @@
+      <Icon src={SearchIcon} />
+      <EmptyText>아직 알림이 없어요...</EmptyText>
+    </Container>
+  );
```
메시지 상수화 해야할 것 같아요!

Donggggg commented on client/src/components/BubbleModal/Notification/Message.tsx:
```diff
@@ -0,0 +1,46 @@
+      case 'MENTORING_REQUEST':
+        return applyText(data.senderName, '그룹 멘토 요청을');
+    }
+  };
```
utils 폴더로 분리하면 좋을 것 같아요!

Donggggg commented on client/src/components/BubbleModal/index.tsx:
```diff
@@ -24,7 +29,20 @@ function BubbleModal({ type, items, headerVisibility }: Props): JSX.Element {
   }
 
-  return <NotificationModalContainer></NotificationModalContainer>;
+  if (notificationList.length === 0)
+    return (
+      <NotificationModalContainer>
+        <EmptyItem />
+      </NotificationModalContainer>
+    );
+  
+  return (
+    <NotificationModalContainer>
+      {notificationList.map((notificationData, idx) => (
+        <ListItem key={notificationData.id} data={notificationData} idx={idx} />
+      ))}
+    </NotificationModalContainer>
+  );
```
하나의 return문으로 작성하면 더 좋을 것 같아요!

yuchanleeme commented on client/src/components/BubbleModal/Notification/Message.tsx:
```diff
@@ -0,0 +1,46 @@
+
+function Message({ data }: Props): JSX.Element {
+  const getText = () => {
+    switch (data.type) {
```
여기 밑에 있는 값들은 함수로 따로 분리해서 하는것보다 템플릿 리터럴로 정확하게 명시해주는게 더 가독성이 좋을것 같습니다!

yuchanleeme commented on client/src/components/BubbleModal/Notification/ListItem.tsx:
```diff
@@ -0,0 +1,103 @@
+        break;
+    }
+
+    newList[idx].readChk = 1;
```
0,1 보다는 true/false로 하면 좋을것 같습니다!

ChanYangYu committed:
[FE] refactor: enum, const 적용


- 데이터 enum, const 사용
- 텍스태 생성 로직 util로 분리
- 알림창 색상 변경
- issue (#208)

ChanYangYu committed:
[BE/FE] feat: 알람 API 연결


- 알람 API 연결 및 서버 API 로직 오류 수정
- 서버데이터 타입 수정
- 1초에 한번씩 알람 가져오도록 interval 설정
- issue (#208)

ChanYangYu committed:
[BE] test: 쿼리 변경에 따른 테스트 수정

- issue (#208)
